Libraries/LibWeb/Painting/ViewportPaintable.cpp

```diff
@@ -315,14 +315,17 @@ void ViewportPaintable::recompute_selection_states(DOM::Range& range)
 
     // 4. Mark the selection end node as End (if text) or Full (if anything else).
     if (auto* paintable = end_container->paintable()) {
-        if (is<DOM::Text>(*end_container))
+        if (is<DOM::Text>(*end_container) || end_container->is_ancestor_of(start_container))
             paintable->set_selection_state(SelectionState::End);
         else
-            paintable->set_selection_state(SelectionState::Full);
+            paintable->for_each_in_inclusive_subtree([&](auto& layout_node) {
+                layout_node.set_selection_state(SelectionState::Full);
+                return TraversalDecision::Continue;
+            });
```

Comment:
Nit: This could do with {}

Reply:
Done.
Hadn't added them at first because they weren't there before.